### PR TITLE
Feature/increase test coverage

### PR DESCRIPTION
**Pull request by CristianDRM**
## Goal
Increase test coverage for Core.swift and Container.swift.
Fix events triggered when the activeContainer was changed.

### CODE REVIEW

**Comment by CristianDRM** [comment]
[[Nova UI] Aplicar testes na estrutura atual](https://trello.com/c/HTn1p4Ny/347-nova-ui-aplicar-testes-na-estrutura-atual)

File: Tests/ContainerTests.swift [hunk 1]
```diff
@@ -27,6 +27,14 @@ class ContainerTests: QuickSpec {
                     expect(container.playback?.pluginName) == "AVPlayback"
                 }
 
+                it("should has background color `clear`") {
```
Comment (leandroalonso) [inline Tests/ContainerTests.swift]: We don't need to necessarily change this right now, but keep in mind: using `should` in tests is a bad practice.

Quoting [from here](http://www.betterspecs.org/#should):

> Do not use should when describing your tests. Use the third person in the present tense.

Comment (CristianDRM) [reply Tests/ContainerTests.swift]: Ok, i will read this on future to improve my unit tests.

File: Tests/CoreTests.swift [hunk 2]
```diff
@@ -126,12 +142,148 @@ class CoreTests: QuickSpec {
...
+
+                    core.on(InternalEvent.willDestroy.rawValue) { _ in
+                        didCallDidDestroy = true
+                    }
+                    core.destroy()
+
+                    expect(didCallDidDestroy).toEventually(beTrue())
+                }
+
+                it("Should remove listeners") {
```
Comment (leandroalonso) [inline Tests/CoreTests.swift]: This test is doing more than what the description informs: it checks for the correctness of the `listenTo` and then check if the listener was removed. We could break this for better legibility.

File: Tests/CoreTests.swift [hunk 3]
```diff
@@ -126,12 +142,148 @@ class CoreTests: QuickSpec {
...
+
+                    core.on(InternalEvent.didChangeActiveContainer.rawValue)   { userInfo in
+                        didCallChangeActiveContainer = true
+                    }
+                    core.activeContainer = Container()
+
+                    expect(didCallChangeActiveContainer).toEventually(beTrue())
+                }
+
+                it("Should listen willChangePlayback and trigger willChangeActivePlayback for new Container") {
```
Comment (leandroalonso) [inline Tests/CoreTests.swift]: This test might give a false positive. Imagine the scenario where no `willChangePlayback` is triggered, but instead, two `willChangeActivePlayback` was triggered. The test will pass, but the behaviour is incorrect.

We should check if every callback was called, not the number of times.

File: Tests/CoreTests.swift [hunk 4]
```diff
@@ -126,12 +142,148 @@ class CoreTests: QuickSpec {
...
+
+                    core.activeContainer = container
+                    core.activeContainer?.playback = AVFoundationPlayback()
+                    core.activeContainer = Container() // prevent events to be called for old container
+
+                    expect(countOfCallEvents).toEventually(equal(2))
+                }
+
+                it("Should listen didChangePlayback and trigger didChangeActivePlayback for new Container") {
+                    let core = Core()
```
Comment (leandroalonso) [inline Tests/CoreTests.swift]: Same as before, it could give a false positive.

File: Tests/CoreTests.swift [hunk 5]
```diff
@@ -126,12 +142,148 @@ class CoreTests: QuickSpec {
...
+
+                    core.activeContainer = container
+                    core.activeContainer?.playback = AVFoundationPlayback()
+                    core.activeContainer = Container() // prevent events to be called for old container
+
+                    expect(countOfCallEvents).toEventually(equal(2))
+                }
+
+                it("Should'nt trigger events twice when container and playback was changed") {
+                    let core = Core()
```
Comment (leandroalonso) [inline Tests/CoreTests.swift]: Again. :)

File: Tests/CoreTests.swift (hunk 3 above)
Comment (CristianDRM) [reply Tests/CoreTests.swift]: I remove the trigger test, cuz doesnt make sense to test this action here.

**Pull request merged**